homeassistant/components/google_assistant/trait.py

```diff
@@ -2070,3 +2073,59 @@ def query_attributes(self):
             "activityState": self.activity_lookup.get(self.state.state, "INACTIVE"),
             "playbackState": self.playback_lookup.get(self.state.state, "STOPPED"),
         }
+
+
+@register_trait
+class ChannelTrait(_Trait):
+    """Trait to get media playback state.
+
+    https://developers.google.com/actions/smarthome/traits/channel
+    """
+
+    name = TRAIT_CHANNEL
+    commands = [COMMAND_SELECT_CHANNEL]
+
+    @staticmethod
+    def supported(domain, features, device_class, _):
+        """Test if state is supported."""
+        if (
+            domain == media_player.DOMAIN
+            and (features & media_player.SUPPORT_PLAY_MEDIA)
+            and device_class == media_player.DEVICE_CLASS_TV
+        ):
+            return True
+
+        return False
+
+    def sync_attributes(self):
+        """Return attributes for a sync request."""
+        return {"availableChannels": [], "commandOnlyChannels": True}
+
+    def query_attributes(self):
+        """Return channel query attributes."""
+        return {}
+
+    async def execute(self, command, data, params, challenge):
+        """Execute an setChannel command."""
+        if command == COMMAND_SELECT_CHANNEL:
+            channel_number = params.get("channelNumber")
+        else:
+            raise SmartHomeError(ERR_NOT_SUPPORTED, "Unsupported command")
+
+        if not channel_number:
+            raise SmartHomeError(
+                ERR_NO_AVAILABLE_CHANNEL,
+                "Channel is not available",
+            )
+
+        await self.hass.services.async_call(
+            media_player.DOMAIN,
+            media_player.SERVICE_PLAY_MEDIA,
+            {
+                ATTR_ENTITY_ID: self.state.entity_id,
+                media_player.ATTR_MEDIA_CONTENT_ID: channel_number,
```

Comment:
This is a bit strange assumption that the channel number is also the id of the channel. I know there might not be a better way now. But for example Philips TV doesn't support this. The same number exists on multiple favorite lists so the media is have to take this into account.

I have no better idea at the moment though :(

Reply:
But given that the media type is set to CHANNEL, it should not look at favorite lists right?

Reply:
Well at least on a Philips tv, you need to tell the tv which list a channel belong to.

When we browse say a channel list in the media browser, we also need to indicate which channel list it's on when we issue a play_media command. The tv will select the channel on that list (satellite, cable , terrestrial, favorites X) as well as switch to that favorite list. 


Also it's identifiers is not a channel index. Ie its like a primary key in a database.

The fact that it has an associated channel number is different from it's identifer. I see it a bit like entity_id vs unique_id.

When we browse media, we need something unique to identify the exact thing. Here we are a bit more vague in saying some channel index. User/Machine

It can likely be handled in the integration, but would need special handling for every integration.